packages/preferences/src/browser/preferences-tree-widget.ts

```diff
@@ -450,7 +454,7 @@ export class PreferencesEditorsContainer extends DockPanel {
 
         let uri = preferenceUri;
         if (preferenceUri.scheme === UserStorageUri.SCHEME && homeUri) {
-            uri = homeUri.resolve(THEIA_USER_STORAGE_FOLDER).resolve(preferenceUri.path);
+            uri = homeUri.resolve(await this.envServer.getDataFolderName()).resolve(preferenceUri.path);
```

Comment:
we should get rid of `homeUri` here and use `getUserDataFolder` instead, `homeUri` breaks abstraction by assuming that user home is `this.fileSystem.getCurrentUserHome`

Reply:
I wonder should we deprecate `FileSystem.getCurrentUserHome` in favour of `EnvVariableServer.getUserHomeFolder`. I think it would be more correct conceptually and prohibit breaking the abstraction as above.

cc @kittaakos @svenefftinge

Reply:
let's deprecate, it does not make sense to have 2 APIs for them same thing. No need to change clients right now, just add @deprecate annotation which says that `EnvVariableServer.getUserHomeFolder` should be used instead